RR.App.Tests/Controllers/HRIS/EmployeeSalaryDetailsControllerUnitTests.cs

```diff
@@ -0,0 +1,281 @@
+
+    public EmployeeSalaryDetailsControllerUnitTest()
+    {
+        _dbMock = new Mock<IUnitOfWork>();
+        _salaryService = new Mock<IEmployeeSalarayDetailsService>();
+        _employeeController = new EmployeeSalaryDetailsController(_salaryService.Object);
+
+        _employeeSalaryDetailsDto = new EmployeeSalaryDetailsDto
+        {
+            Id = 1,
+            EmployeeId = 1,
+            Salary = 2000,
+            MinSalary = 1500,
+            MaxSalary = 3000,
+            Remuneration = 2500,
+            Band = EmployeeSalaryBand.Level1,
+            Contribution = null
+        };
+
+        _salaries = new List<EmployeeSalaryDetailsDto> { _employeeSalaryDetailsDto };
+    }
+
+    [Fact]
+    public async Task SaveEmployeeSalaryValidInputReturnsOkResult()
+    {
+        var employeeSalaryDetailsServiceMock = new Mock<IEmployeeSalarayDetailsService>();
+        var employeeServiceMock = new Mock<IEmployeeService>();
+        var controller = new EmployeeSalaryDetailsController(employeeSalaryDetailsServiceMock.Object);
+
+        var employeeSalaryDetailsDto = new EmployeeSalaryDetailsDto
+        {
+            Id = 1,
+            EmployeeId = 1,
+            Salary = 2000,
+            MinSalary = 1500,
+            MaxSalary = 3000,
+            Remuneration = 2500,
+            Band = EmployeeSalaryBand.Level1,
+            Contribution = null
+        };
+
+        employeeServiceMock.Setup(x => x.GetEmployeeById(employeeSalaryDetailsDto.EmployeeId))
+                           .ReturnsAsync(EmployeeTestData.EmployeeDto);
+
+        employeeSalaryDetailsServiceMock.Setup(x => x.SaveEmployeeSalary(It.IsAny<EmployeeSalaryDetailsDto>()))
+                                        .ReturnsAsync(employeeSalaryDetailsDto);
+
+        var result = await controller.AddEmployeeSalary(employeeSalaryDetailsDto);
+
+        var createdAtActionResult = Assert.IsType<CreatedAtActionResult>(result);
+        Assert.Equal(nameof(controller.AddEmployeeSalary), createdAtActionResult.ActionName);
+        Assert.Equal(employeeSalaryDetailsDto.EmployeeId, createdAtActionResult.RouteValues["employeeId"]);
+        Assert.Equal(employeeSalaryDetailsDto, createdAtActionResult.Value);
+    }
+
+    [Fact]
+    public async Task SaveEmployeeSalaryExceptionThrownReturnsNotFoundWithMessage()
+    {
+        var employeeSalaryDetailsServiceMock = new Mock<IEmployeeSalarayDetailsService>();
+        var employeeServiceMock = new Mock<IEmployeeService>();
+        var controller = new EmployeeSalaryDetailsController(employeeSalaryDetailsServiceMock.Object);
+
+        var employeeSalaryDetailsDto = new EmployeeSalaryDetailsDto
+        {
+            Id = 1,
+            EmployeeId = 1,
+            Salary = 2000,
+            MinSalary = 1500,
+            MaxSalary = 3000,
+            Remuneration = 2500,
+            Band = EmployeeSalaryBand.Level1,
+            Contribution = null
+        };
```

Comment:
It is preferred to store the dto in the test data file so the unit test file can be cleaned up

Reply:
Makes sense. I'll make use of the EmployeeSalaryDetailsTestData instead.